dolistore/modules/blockmysales/blockmysales.php

```diff
@@ -1,2275 +1,2277 @@
-
-				$result = Db::getInstance()->Execute($query);
-				if ($result === false) die(Tools::displayError('Invalid loadLanguage() SQL query! : '.$query));
-
-				// Add tag description of product
-				$tags=preg_split('/[\s,]+/', $product['keywords']);
-				foreach($tags as $tag)
-				{
-					$id_tag=0;
-
-					// Search existing tag
-					$query = 'SELECT `id_tag` FROM `'._DB_PREFIX_.'tag` WHERE `id_lang` = \''.$id_lang.'\' AND `name` = \''.addslashes($tag).'\' ';
-					$result = Db::getInstance()->ExecuteS($query);
-					if ($result === false) die(Tools::displayError('Invalid loadLanguage() SQL query!: '.$query));
-					foreach ($result AS $row)
-					{
-						$id_tag = $row['id_tag'];
-						//self::prestalog("tag id for id_lang ".$id_lang.", name ".$tag." is ".$id_tag);
-					}
-
-					if (empty($id_tag))
-					{
-						$query = "INSERT INTO `"._DB_PREFIX_."tag` (`id_lang`, `name`) VALUES ('".$id_lang."', '".addslashes($tag)."')";
-						$result = Db::getInstance()->Execute($query);
-						//if ($result === false) die(Tools::displayError('Invalid loadLanguage() SQL query! : '.$query));
-
-						$id_tag = Db::getInstance()->Insert_ID();
-						//self::prestalog("We created tag for id_lang ".$id_lang.", name ".$tag.", id is ".$id_tag);
-					}
-
-					if (! empty($id_tag) && $id_tag > 0)
-					{
-						// Add tag link
-						$query = "INSERT INTO `"._DB_PREFIX_."product_tag` (`id_product`, `id_tag`) VALUES ('".$id_product."', '".$id_tag."')";
-						$result = Db::getInstance()->Execute($query);
-
-						//self::prestalog("We insert link id_product ".$id_product.", id_tag ".$id_tag);
-					}
-				}
-			}
-
-			$product_file_name = Tools::getValue('product_file_name');
-			$product_file_path = Tools::getValue('product_file_path');
-			$newfile = ($product_file_path ? 1 : 0);
-
-			$nbdaysaccessible = Configuration::get('BLOCKMYSALES_NBDAYSACCESSIBLE');
-			$nb_days_accessible = (Tools::isSubmit('nb_days_accessible') ? Tools::getValue('nb_days_accessible') : (!empty($nbdaysaccessible) ? $nbdaysaccessible : 365));
-
-			//mise en base du lien avec le produit telechargeable
-			if ($newfile)
-			{
-				$product_file_newname = basename($product_file_path);
-
-				$query = 'DELETE FROM `'._DB_PREFIX_.'product_download` WHERE `id_product` = '.$id_product.';';
-				$result1 = Db::getInstance()->Execute($query);
-
-				$query = 'INSERT INTO `'._DB_PREFIX_.'product_download` (`id_product`, `display_filename`, `filename`, `date_add`, `date_expiration`, `nb_days_accessible`, `nb_downloadable`, `active`) VALUES (
-						'.$id_product.', "'.$product_file_name.'", "'.$product_file_newname.'", "'.$dateNow.'", "0000-00-00 00:00:00", "'.$nb_days_accessible.'", 0, 1)';
-				//self::prestalog("A new file is asked: We add it into product_download query=".$query);
-				$result = Db::getInstance()->Execute($query);
-				if ($result === false) die(Tools::displayError('Invalid loadLanguage() SQL query! : '.$query));
-			}
-			else
-			{
```

Comment:
Can you confirm it is 
$iso_code = $this->context->language->iso_code;
and not
$iso_code = $context->language->iso_code;